File: XBRL/xbrl_parse.scala
import scala.xml.XML

val xml = XML.loadFile(args(0))

val values = xml \ "OperatingIncome" filter(n => (n \ "@contextRef").text == "CurrentYearConsolidatedDuration")

//OK
//val values = xml \ "OperatingIncome" filter(_.attribute("contextRef").mkString == "CurrentYearConsolidatedDuration")

//NG
//val values = xml \ "OperatingIncome" filter((_ \ "@contextRef").text == "CurrentYearConsolidatedDuration")

values.map(_.text).foreach(println)

File: Haskell/HaXml/xbrl_parse.hs
import System
import Text.XML.HaXml
import Text.XML.HaXml.Posn
import Text.XML.HaXml.Util
import Text.XML.HaXml.Xtract.Parse

main = do
	args <- getArgs
	contents <- readFile $ head args

	let
		Document _ _ root _ = xmlParse "" contents
		cont = CElem root noPos

	mapM_ (print . tagTextContent) $ xtract id "//jpfr-t-cte:OperatingIncome[@contextRef='CurrentYearConsolidatedDuration']" cont

-- ˆÈ‰º‚Å‚à‰Â
--	let Elem n _ cont = root
--	mapM_ (print . tagTextContent) $ concatMap (xtract id "//jpfr-t-cte:OperatingIncome[@contextRef='CurrentYearConsolidatedDuration']") cont

